olp-cpp-sdk-core/src/http/curl/NetworkCurl.cpp

```diff
@@ -1058,5 +1071,25 @@ void NetworkCurl::Run() {
   OLP_SDK_LOG_DEBUG(kLogTag, "Thread exit, this=" << this);
 }
 
+void NetworkCurl::GetTraficData(CURL* handle, uint64_t& uploadBytes,
+                                uint64_t& downloadBytes) {
+  long headersSize;
+  double lengthDownloaded;
+  if (curl_easy_getinfo(handle, CURLINFO_HEADER_SIZE, &headersSize) ==
+          CURLE_OK &&
+      curl_easy_getinfo(handle, CURLINFO_SIZE_DOWNLOAD, &lengthDownloaded) ==
+          CURLE_OK) {
+    downloadBytes = lengthDownloaded + headersSize;
```

Comment:
Maybe a fallback should be in place in case curl_easy_getinfo is not delivering responses, e.g. mark the payload size and the headers size in RequestHandle.count before you send the request.

Reply:
Do you mean that we should count upload bytes before curl_multi_perform?

Reply:
Just in case. We should still use curl_easy_getinfo, but if those calls fail we still have a callback.

Reply:
Theoretically curl_easy_getinfo should always be successful  if all parameters ok. On send headers are set internally by curl_easy_setopt. So we can count only user headers. The same on response. So maybe we should do fallback for all platforms, if upload/download is 0, but data received is not 0.